java/arcs/core/policy/Policy.kt

```diff
@@ -18,21 +18,39 @@ import arcs.core.data.Capability
 /** Defines a data usage policy. See [PolicyProto] for the canonical definition of a policy. */
 data class Policy(
     val name: String,
-    val description: String,
     val egressType: EgressType,
-    val targets: List<PolicyTarget>,
-    val configs: Map<String, PolicyConfig>,
-    val annotations: List<Annotation>
-)
+    val description: String = "",
+    val targets: List<PolicyTarget> = emptyList(),
+    val configs: Map<String, PolicyConfig> = emptyMap(),
+    val annotations: List<Annotation> = emptyList()
+) {
+    /** The name of the egress particle that matches this policy. */
+    val egressParticleName = "Egress_$name"
+
+    /** The set of all fields (includes nested fields). */
+    val allFields: Set<PolicyField> = computeAllFields()
+
+    /** The set of all redaction labels mentioned in the policy. */
+    val allRedactionLabels: Set<String> = allFields.flatMap { it.redactedUsages.keys }.toSet()
+
+    private fun computeAllFields(): Set<PolicyField> {
+        fun getAllFields(field: PolicyField): List<PolicyField> {
+            return listOf(field) + field.subfields.flatMap { getAllFields(it) }
+        }
+        return targets.flatMap { target ->
+            target.fields.flatMap { getAllFields(it) }
+        }.toSet()
```

Comment:
Mostly a clarification, do we need to convert to a `Set`?

Reply:
Probably not actually. Done.